### PR TITLE
Add payment mode selection with conference for organisers

### PR DESCRIPTION
**Pull request by rishabhs95**
add capability for organisers to select payment modes for their conferences as Online/Offline.

### CODE REVIEW

**Comment by rishabhs95** [comment]
will replace dropdown with better UI alternative as suggested by hernan

File: db/migrate/20160531152256_add_payment_mode_to_conferences.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddPaymentModeToConferences < ActiveRecord::Migration
+  def change
+    add_column :conferences, :payment_mode, :string, default: 'Offline'
```
Comment (lagartoflojo) [inline db/migrate/20160531152256_add_payment_mode_to_conferences.rb]: Do we want the `payment_mode` column to ever be `NULL`?

Comment (lagartoflojo) [inline db/migrate/20160531152256_add_payment_mode_to_conferences.rb]: Also, how about using Enums?
I think it fits this use case pretty nicely.
http://api.rubyonrails.org/v4.1.0/classes/ActiveRecord/Enum.html

File: app/models/conference.rb [hunk 2]
```diff
@@ -57,6 +57,9 @@ class Conference < ActiveRecord::Base
   validates_format_of :short_title, with: /\A[a-zA-Z0-9_-]*\z/
   validates :registration_limit, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
+  validates :payment_mode, presence: true, inclusion: {in: %w(Online Offline),
```
Comment (lagartoflojo) [inline app/models/conference.rb]: I would go for lower-case `online` and `offline`

File: db/migrate/20160531152256_add_payment_mode_to_conferences.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddPaymentModeToConferences < ActiveRecord::Migration
+  def change
+    add_column :conferences, :payment_mode, :string, default: 'offline'
```
Comment (lagartoflojo) [inline db/migrate/20160531152256_add_payment_mode_to_conferences.rb]: Hmm now that I think about it, "payment method" is better than payment mode :-\
http://www.google.com/trends/explore#q=%22payment%20method%22%2C%20%22payment%20mode%22&cmpt=q&tz=Etc%2FGMT-2

Comment (rishabhs95) [inline db/migrate/20160531152256_add_payment_mode_to_conferences.rb]: Ok, will change the name to `payment_method`

File: app/models/conference.rb (hunk 2 above)
Comment (rishabhs95) [inline app/models/conference.rb]: Yes, I've made the changes for it.

File: db/migrate/20160531152256_add_payment_mode_to_conferences.rb (hunk 1 above)
Comment (rishabhs95) [inline db/migrate/20160531152256_add_payment_mode_to_conferences.rb]: I am not using `payment_mode` to ever be `NULL`, so I've added `null: false`  on `payment_method` as well.

File: spec/models/conference_spec.rb [hunk 4]
```diff
@@ -1545,6 +1545,18 @@
...
+    end
+
+    it 'is valid with payment mode as Online' do
+      should allow_value('online').for(:payment_method)
+    end
+
+    it 'is valid with payment mode as Offline' do
+      should allow_value('offline').for(:payment_method)
+    end
+
```
Comment (kalabiyau) [inline spec/models/conference_spec.rb]: I would also test the message of the validator to be consistent.

Comment (rishabhs95) [inline spec/models/conference_spec.rb]: I have updated the validator tests for custom messages as you suggested

File: app/models/conference.rb [hunk 5]
```diff
@@ -57,6 +57,9 @@ class Conference < ActiveRecord::Base
   validates_format_of :short_title, with: /\A[a-zA-Z0-9_-]*\z/
   validates :registration_limit, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
+  validates :payment_method, presence: true, null: false, inclusion: {in: %w(online offline),
```
Comment (hennevogel) [inline app/models/conference.rb]: This type of data is called [enum](http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html)

File: app/views/admin/conference/edit.html.haml [hunk 6]
```diff
@@ -23,4 +23,13 @@
         = f.input :end_date, :as => :string, :input_html => { :id => "conference-end-datepicker", :readonly => "readonly"  }
       = f.inputs name: "Registrations" do
         = f.input :registration_limit, as: :number, in: 0..9999, hint: "Limit the number of registrations to the conference (0 no limit)"
+      = f.inputs name: "Tickets" do
```
Comment (hennevogel) [inline app/views/admin/conference/edit.html.haml]: We use [Bootstrap-Switch-Checkbox](https://github.com/openSUSE/osem/wiki/Bootstrap-Switch-Checkbox) for this kind of input

**Comment by lagartoflojo** [comment]
@hennevogel I think we can close this PR, as we are not going to offer the "offline" payment option, correct?

**Comment by hennevogel** [comment]
true!

**Pull request closed without merging**